src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/Reliability/ITransientErrorDetectionStrategy.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data.SqlClient.Reliability
+{
+    /// <summary>
+    /// Defines an interface that must be implemented by custom components responsible for detecting specific transient conditions.
+    /// </summary>
+    public interface ITransientErrorDetectionStrategy
+    {
+        /// <summary>
+        /// Determines whether the specified exception represents a transient failure that can be compensated by a retry.
+        /// </summary>
+        /// <param name="ex">The exception object to be verified.</param>
+        /// <returns>true if the specified exception is considered as transient; otherwise, false.</returns>
+        bool IsTransient(Exception ex);
+
+        /// <summary>
+        /// Determines whether the specified exception represents a transient failure that can be compensated by a retry.
+        /// </summary>
+        /// <returns>true if the specified exception is considered as transient; otherwise, false.</returns>
+        List<int> RetriableErrors
```

Comment:
`List<int>`  should not occur in a public interface. Likely, some immutable value should be returned. Also, it seems that the idea that errors are identified by a set of integers is specific to SqlClient. Other providers might identify errors though other types such as long, enum, string or large ranges of integers.